hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAssociateToAccountHandler.java

```diff
@@ -170,18 +180,22 @@ private boolean isTotalNumTokenRelsWithinMax(
      * Method that checks if the number of token associations for the given account is within the
      * allowable limit set by the config (if the limit is enabled)
      *
-     * @return true if tokenAssociationsLimited is false or if the number of token associations is
+     * @return true if tokenAssociationsLimited is false or if the unlimitedAutoAssociations is true
+     * or the number of token associations is
      * within the allowed maxTokensPerAccount
      */
     private boolean maxAccountAssociationsAllowTokenRels(
             @NonNull final TokensConfig config,
             @NonNull final EntitiesConfig entitiesConfig,
+            @NonNull final ContractsConfig contractsConfig,
             @NonNull final Account account,
             @NonNull final List<TokenID> tokenIds) {
         final var numAssociations = requireNonNull(account).numberAssociations();
         final var tokenAssociationsLimited = entitiesConfig.limitTokenAssociations();
         final var maxTokensPerAccount = config.maxPerAccount();
-        return !tokenAssociationsLimited || (numAssociations + tokenIds.size() <= maxTokensPerAccount);
+        return !tokenAssociationsLimited
+                || ((numAssociations + tokenIds.size() <= maxTokensPerAccount)
+                        && contractsConfig.unlimitedAutoAssociations());
```

Comment:
Shouldn't this be 
`return !tokenAssociationsLimited || contractsConfig.unlimitedAutoAssociations()  ||
            (numAssociations + tokenIds.size() <= maxTokensPerAccount);`
So its || instead of &&

Reply:
Discussed offline, we shouldn't override the tokenAssociationsLimited flag with the feature toggle.